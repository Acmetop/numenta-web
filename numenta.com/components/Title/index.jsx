// Numenta company website source code. Copyright Â© 2016 Numenta.
// Full details in LICENSE.txt, or contact us at <http://numenta.com>.
// This program is free software: you can redistribute it and/or modify it under
// the terms of the GNU Affero General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version. This program is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
// General Public License for more details. You should have received a copy of
// the GNU Affero General Public License along with this program. If not, see
// <https://www.gnu.org/licenses/agpl.html>.

import React from 'react'

import styles from './index.css'


const Title = ({children, headline}) => {
  const isHeadline = headline === true
  const className = isHeadline ? styles.headline : styles.title

  return (
    <h1 className={className}>
      {children}
    </h1>
  )
}

Title.propTypes = {
  children: React.PropTypes.node.isRequired,
  headline: React.PropTypes.bool.isRequired,
}

Title.defaultProps = {
  headline: false,
}

export default Title
