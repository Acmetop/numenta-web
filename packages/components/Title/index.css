/**
 * Numenta shared web components
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

._base {
  composes: black from 'tachyons';
  composes: fw6 from 'tachyons';
  composes: lh-title from 'tachyons';
  composes: mv1 from 'tachyons';
  composes: pa0 from 'tachyons';
}


.headline {
  composes: f2 from 'tachyons';
  composes: _base;
}

.title {
  composes: f3 from 'tachyons';
  composes: _base;
}
