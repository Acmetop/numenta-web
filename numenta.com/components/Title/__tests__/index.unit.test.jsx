import React from 'react'
import renderer from 'react-test-renderer'

import Title from '../../Title'


describe('Title React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Title>
        Important Text
      </Title>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
