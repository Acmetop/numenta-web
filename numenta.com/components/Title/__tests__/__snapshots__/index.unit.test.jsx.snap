exports[`Title React component Renders correctly 1`] = `
<h1
  className="title">
  Important Text
</h1>
`;
